lib/hotwire/spark/middleware.rb

```diff
@@ -46,10 +46,15 @@ def inject_options(html)
         html.sub("</head>", "#{logging_option}</head>")
       else
         html
-      end
+      end.sub("</head>", "#{html_reload_strategy_option}</head>")
```

Comment:
Now that we have more than one option I think it would be nicer if we separated:

- The replacement action.
- From the logic composing the list of options

```ruby
    def inject_options(html)
      html.sub("</head>", "#{options}</head>")
    end

    def options
      [ logging_option_option, reload_method_option ].compact.join("\n")
    end

    def logging_option_option
      view_helpers.tag.meta(name: "hotwire-spark:logging", content: "true") if Hotwire::Spark.logging
    end

    def reload_method_option
      view_helpers.tag.meta(name: "hotwire-spark:html-reload-strategy", content: Hotwire::Spark.html_reload_method)
    end
```

Reply:
I applied your suggestion 👍 

An alternative/different approach could be to inject only a single meta tag with the complete JSON configuration:

```ruby
view_helpers.tag.meta(name: "hotwire-spark:configuration", content: { ... }.to_json)
```

```js
document.addEventListener("DOMContentLoaded", function() {
  const metaConfig = document.querySelector(`meta[name="hotwire-spark:configuration"]`)
  HotwireSpark.config = JSON.parse(metaConfig.content)
})
```